test/instruments_test.go

```diff
@@ -14,120 +14,201 @@ import (
 )
 
 func TestCreateAndGetInstrument(t *testing.T) {
-
 	cardTokenResponse := RequestCardToken(t)
 	response := createTokenInstrument(t, cardTokenResponse)
-	assert.Equal(t, instruments.Card, response.Type)
-	assert.NotEmpty(t, response.Id)
-	assert.NotEmpty(t, response.Fingerprint)
-	assert.NotEmpty(t, response.ExpiryMonth)
-	assert.NotEmpty(t, response.ExpiryYear)
-	assert.NotEmpty(t, response.Scheme)
-	assert.NotEmpty(t, response.Last4)
-	assert.NotEmpty(t, response.Bin)
-	assert.NotEmpty(t, response.CardType)
-	assert.NotEmpty(t, response.CardCategory)
-	assert.NotEmpty(t, response.ProductId)
-	assert.NotEmpty(t, response.ProductType)
-
-	getResponse, err := DefaultApi().Instruments.Get(response.Id)
-	assert.Nil(t, err)
-	assert.NotNil(t, response)
-	getCardInstrumentResponse := getResponse.GetCardInstrumentResponse
-	assert.Equal(t, instruments.Card, getCardInstrumentResponse.Type)
-	assert.NotEmpty(t, getCardInstrumentResponse.Id)
-	assert.NotEmpty(t, getCardInstrumentResponse.Fingerprint)
-	assert.NotEmpty(t, getCardInstrumentResponse.AccountHolder)
-	assert.NotEmpty(t, getCardInstrumentResponse.ExpiryMonth)
-	assert.NotEmpty(t, getCardInstrumentResponse.ExpiryYear)
-	assert.NotEmpty(t, getCardInstrumentResponse.Name)
-	assert.NotEmpty(t, getCardInstrumentResponse.Scheme)
-	assert.NotEmpty(t, getCardInstrumentResponse.Last4)
-	assert.NotEmpty(t, getCardInstrumentResponse.Bin)
-	assert.NotEmpty(t, getCardInstrumentResponse.CardType)
-	assert.NotEmpty(t, getCardInstrumentResponse.CardCategory)
-	assert.NotEmpty(t, getCardInstrumentResponse.ProductId)
-	assert.NotEmpty(t, getCardInstrumentResponse.ProductType)
+
+	cases := []struct {
+		name       string
+		responseId string
+		checker    func(*nas.GetInstrumentResponse, error)
+	}{
+		{
+			name:       "when get a created instrument then return it",
```

Comment:
Check the names for the cases in other tests files like customers_test and adapt them so that they are more readable

Reply:
This one would be better as "when fetching a valid instrument then return instrument data"

Reply:
Please check the names for all the cases

Reply:
👍